src/PrestaShopBundle/Entity/Repository/AttributeRepository.php

```diff
@@ -85,6 +89,46 @@ public function findByLangAndShop($idLang, $idShop)
         return $attributeGroups;
     }
 
+    /**
+     * Asserts that attribute exists in all the provided shops.
+     * If at least one of them is missing in any shop, it throws exception.
+     *
+     * @param AttributeId[] $attributeIds
+     * @param ShopId[] $shopIds
+     *
+     * @throws ShopAssociationNotFound
+     */
+    public function assertExistsInEveryShop(array $attributeIds, array $shopIds): void
```

Comment:
Same suggestion it seems like it can easily be moved in the other repository Let's avoid this Doctrine EntityRepository as much as we can 😅